hazelcast/src/main/java/com/hazelcast/sql/SqlResult.java

```diff
@@ -53,17 +78,28 @@
      * The iterator may be requested only once.
      *
      * @return iterator
-     * @throws IllegalStateException if the method is invoked more than once
+     * @throws IllegalStateException if the method is invoked more than once or
+     *    if this result doesn't have rows (i.e. when {@link #isUpdateCount()}
+     *    returns {@code true})
      * @throws SqlException in case of an SQL-related error condition
      */
     @Nonnull
     @Override
     Iterator<SqlRow> iterator();
 
     /**
-     * Release the resources associated with the query result.
+     * Returns the number of rows updated by the execution.
+     *
+     * @throws IllegalStateException if this result doesn't represent an update
```

Comment:
In JDBC, if the result is a result set, then `-1` is returned. But we throw an exception. 

Not advocating for any of these approaches, my question is - do we have any specific reason to prefer one over the other? One minor advantage of the `-1` marker, is that we may use it instead of `isUpdateCount` to check for result type.

Reply:
I think it's the symmetry between `iterator()` and `updateCount()` - they throw if called for an incorrect result type. Also my point is that one should know what should be the result type for a particular query. We even advocate for not calling `isUpdateCount` but directly calling the method to get the particular type of result. `isUpdateCount` is meant for a generic query UI.

Reply:
Also in Jet we won't implement update count for DML. In that case the `-1` return value will mean that the update count is unknown, not that there are rows. We would need two return values, e.g.:
- -1: unknown update count
- -2: not a result with update count, but with rows

Reply:
Makes sense, thank you